libbeat/autodiscover/template/config.go

```diff
@@ -71,18 +78,32 @@ func (e Event) GetValue(key string) (interface{}, error) {
 	return val, nil
 }
 
+// SetKubernetesKeystoresRegistry set the k8sKeystoresRegistry of the Mapper object
+func (c *Mapper) SetKubernetesKeystoresRegistry(k8sKeystoresRegistry *keystore.KubernetesKeystoresRegistry) {
+	c.kubernetesKeystoresRegistry = k8sKeystoresRegistry
+}
+
 // GetConfig returns a matching Config if any, nil otherwise
 func (c Mapper) GetConfig(event bus.Event) []*common.Config {
 	var result []*common.Config
-
-	for _, mapping := range c {
+	opts := []ucfg.Option{}
+	if c.keystore != nil {
+		opts = append(opts, ucfg.Resolve(keystore.ResolverWrap(c.keystore)))
+	}
+	if c.kubernetesKeystoresRegistry != nil {
+		k8sKeystore := c.kubernetesKeystoresRegistry.GetK8sKeystore(event)
+		if k8sKeystore != nil {
+			opts = append(opts, ucfg.Resolve(keystore.ResolverWrap(k8sKeystore)))
+		}
+	}
```

Comment:
this adds the k8s keystore at the end of the resolver list. The local keystore in `c.keystore` seems to have priority in case both have the same key included. Maybe we want this the other way around? Do you have a test to confirm the behavior?

Reply:
I don't think we can define a priority here. Some cases may want to use local keystore while others the k8s secrets. I think that local keystore can have higher priority, if we want to set a priority, since it is internal and kind of static while the other one is "external" and dynamic. Also the way the keys are defined is quite unique for k8s keystore, `kubernetes.<namespace>.<secret_name>.<secret_value>` and this makes it highly unlikely to have an overlapping in the keys' names.

Reply:
The 'priority' is implicit by the order you pass the options to go-ucfg. Currently the 'keystore' is passed first, which means it has higher priority if both have the same key.

> I think that local keystore can have higher priority, if we want to set a priority, since it is internal and kind of static while the other one is "external" and dynamic

👍 How about adding a code comment that the order of 'keystore' first is on purpose to give it a higher priority.

> Also the way the keys are defined is quite unique for k8s keystore, kubernetes.<namespace>.<secret_name>.<secret_value> and this makes it highly unlikely to have an overlapping in the keys' names

Might be true. But I can store arbitrary keys in the keystore. Assuming URLs (or database connection string that contains username and password) are stored, it might be possible (in case you know the keys) to redirect the module/input to another known URL via the keystore. Hm, maybe we want to give the k8 keystore a higher priority :/

Reply:
👍